### PR TITLE
[SimpleFormIterator] add TransitionProps prop

### PR DESCRIPTION
**Pull request by cherniavskii**
This PR adds ability to override `CSSTransition` props.

Together with #3223 adds ability to workaround #3220 in user land:

```diff
<ArrayInput source="backlinks">
-  <SimpleFormIterator>
+  <SimpleFormIterator TransitionProps={{ timeout: 400 }}>
    <RichTextInput source="url" />
  </SimpleFormIterator>
</ArrayInput>
```

qxl29-kwkn8

May be worth to decrease default timeout in the future, but adding ability for workaround is a good first step.

### CODE REVIEW

File: packages/ra-ui-materialui/src/form/SimpleFormIterator.js [hunk 1]
```diff
@@ -115,6 +115,7 @@ export class SimpleFormIterator extends Component {
             translate,
             disableAdd,
             disableRemove,
+            TransitionProps,
```
Comment (djhi) [inline packages/ra-ui-materialui/src/form/SimpleFormIterator.js]: Can you rename it to `transitionProps`?

**Comment by cherniavskii** [comment]
From component's API perspective. it's indeed a new feature.
From the other hand, it's the only way to solve #3220 in user land, which is present on current master.

I'll change target branch to `next`, but could you please consider cherrypicking it to `master` branch, so users will have a workaround before react-admin 3.0 release? This change is backwards compatible.

**Comment by djhi** [comment]
Ok, can you wait for @fzaninotto review then before switching branch ?

**Comment by cherniavskii** [comment]
@djhi sure

File: packages/ra-ui-materialui/src/form/SimpleFormIterator.js (hunk 1 above)
Comment (cherniavskii) [reply packages/ra-ui-materialui/src/form/SimpleFormIterator.js]: I used material-ui naming pattern here - see `ExpansionPanel` `TransitionProps` - https://material-ui.com/api/expansion-panel/#props

**Comment by fzaninotto** [comment]
It's indeed a new feature, please PR against `next` rather than `master`. The 2.x branch is feature-freeze.

**Comment by djhi** [comment]
Any news on this one?

**Comment by djhi** [comment]
Thanks!

**Pull request merged**